Added text decoration configuration item

goakly opened the pull request:
### Added ability to use text decorations (italics, bold, underline, etc) via a configuration item called `textDecorations`.

- All decorations are disabled by default.
- A list of possible text decorations can be found [here](https://neovim.io/doc/user/api.html#nvim_set_hl()).
- When using the color palette, text decorations will apply to every color.


_This is my first attempt at Lua so let me know if there is a better way to do this._

m-demare commented:
Hi!
There's actually an option to allow text decorations already (it's called `highlight`)
However, it doesn't work when using the colorpalette, so I think that's a great addition. Could you change that to use the existing option?

goakly commented:
Didn't realize the highlight option could do text decoration. Rewrote so that the color palette could use options set in highlight and I added a section in readme outlining the feature!

goakly commented:
Actually, feel like a section in the readme about text decoration is unnecessary. Removed.

m-demare reviewed (approved):
Nice! Thanks for contributing
